### PR TITLE
feat: Use `issuer` directly

### PR DESCRIPTION
**Pull request by JonaszJestem**

### CODE REVIEW

File: packages/origin-247-certificate/src/onchain-certificate/certificate-operations/claim/batch-claim-certificates.handler.ts [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+    ForbiddenException,
+    HttpException,
+    HttpStatus,
+    NotFoundException
+} from '@nestjs/common';
+import { BlockchainPropertiesService } from '../../blockchain-properties.service';
+import { IBatchClaimCommand } from '../../../types';
+import { BatchCertificateClaim } from '@energyweb/issuer/dist/js/src/blockchain-facade/CertificateBatchOperations';
+
+export class BatchClaimCertificatesHandler {
```
Comment (soanvig) [inline packages/origin-247-certificate/src/onchain-certificate/certificate-operations/claim/batch-claim-certificates.handler.ts:16]: What is the purpose of these handlers (this and others)? Everything is already validated in offchain layer, isn't it?

**Review by soanvig — CHANGES_REQUESTED** [review]
[certificate-updated.handler.ts](https://github.com/energywebfoundation/origin-247-sdk/pull/85/files#diff-ac55f7dd3a38b06933f5efe26b042f4b1b70db70a06468fc7b4452684b8bacdc)

won't be necessary as well right?

File: packages/origin-247-certificate/migrations/1646865380689-AddDeploymentPropertiesIssuer.ts [hunk 2]
```diff
@@ -0,0 +1,13 @@
+import { MigrationInterface, QueryRunner } from 'typeorm';
+
+export class AddDeploymentPropertiesIssuer1646865380690 implements MigrationInterface {
+    name = 'AddDeploymentPropertiesIssuer1646865380690';
+
+    public async up(queryRunner: QueryRunner): Promise<void> {
```
Comment (soanvig) [inline packages/origin-247-certificate/migrations/1646865380689-AddDeploymentPropertiesIssuer.ts:6]: why patryk's zmiany are here

**Review by soanvig — CHANGES_REQUESTED** [review]

File: packages/origin-247-certificate/src/onchain-certificate/blockchain-properties.service.ts [hunk 3]
```diff
@@ -60,9 +56,61 @@ export class BlockchainPropertiesService {
...
+    async wrap(privateKey: string) {
+        const { registry, issuer } = await this.deploymentPropsRepo.get();
+
+        const { web3 } = await this.getProperties();
+
+        const signer: Signer = new Wallet(this.assure0x(privateKey), web3);
+        return {
+            web3,
+            registry: Contracts.factories.RegistryExtendedFactory.connect(registry, signer),
+            privateIssuer: undefined,
```
Comment (soanvig) [inline packages/origin-247-certificate/src/onchain-certificate/blockchain-properties.service.ts:71]: we don't need that, right?

File: packages/origin-247-certificate/src/onchain-certificate/blockchain-properties.service.ts [hunk 4]
```diff
@@ -60,9 +56,61 @@ export class BlockchainPropertiesService {
...
+            activeUser: signer
+        };
     }
 
     private assure0x = (privateKey: string) =>
         privateKey.startsWith('0x') ? privateKey : `0x${privateKey}`;
 }
+
+@Injectable()
+export class BlockchainPropertiesService {
```
Comment (soanvig) [inline packages/origin-247-certificate/src/onchain-certificate/blockchain-properties.service.ts:82]: is this Patryk's change or yours? afair Patryk validated if env are set in the constructor, you don't

File: packages/origin-247-certificate/src/onchain-certificate/certificate-operations/claim/batch-claim-certificates.handler.ts [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+
+        if (claims.length === 0) {
+            throw new BadRequestException('Cannot process empty claims request');
+        }
+
+        const certificates = await this.readModelRepo.getManyByBlockchainCertificateIds(
+            claims.map((c) => c.certificateId)
+        );
+
+        const validatedClaims = claims.map((claim) => {
```
Comment (soanvig) [inline packages/origin-247-certificate/src/onchain-certificate/certificate-operations/claim/batch-claim-certificates.handler.ts:39]: why do we validate anything here, it's validated in offchain already

**Review by soanvig — CHANGES_REQUESTED** [review]

File: packages/origin-247-certificate/test/offchain-certificate/offchain-certificate.e2e-spec.ts [hunk 6]
```diff
@@ -41,7 +41,7 @@ describe('OffchainCertificate', () => {
         await cleanDB();
     });
 
-    it('should issue, transfer, claim and synchronize', async () => {
+    it.only('should issue, transfer, claim and synchronize', async () => {
```
Comment (soanvig) [inline packages/origin-247-certificate/test/offchain-certificate/offchain-certificate.e2e-spec.ts:44]: :)

**Review by soanvig — CHANGES_REQUESTED** [review]

File: packages/origin-247-certificate/migrations/1646865380689-AddDeploymentPropertiesIssuer.ts (hunk 2 above)
Comment (JonaszJestem) [reply packages/origin-247-certificate/migrations/1646865380689-AddDeploymentPropertiesIssuer.ts:6]: this is not Patryk zmiany, this is to store the address of deployed issuer contract

File: packages/origin-247-certificate/src/onchain-certificate/blockchain-properties.service.ts (hunk 3 above)
Comment (JonaszJestem) [reply packages/origin-247-certificate/src/onchain-certificate/blockchain-properties.service.ts]: removed

File: packages/origin-247-certificate/src/onchain-certificate/blockchain-properties.service.ts (hunk 4 above)
Comment (JonaszJestem) [reply packages/origin-247-certificate/src/onchain-certificate/blockchain-properties.service.ts]: I changed it a little bit, the validation is still there

File: packages/origin-247-certificate/src/offchain-certificate/utils/certificate.utils.ts [hunk 7]
```diff
@@ -0,0 +1,8 @@
+import { ICertificateReadModel } from '../types';
+import { CertificateEventType } from '../events/Certificate.events';
+
+export const getCreationTransactionHash = <T>(
```
Comment (soanvig) [inline packages/origin-247-certificate/src/offchain-certificate/utils/certificate.utils.ts:4]: unused

File: packages/origin-247-certificate/src/offchain-certificate/repositories/CertificateReadModel/CertificateReadModelPostgres.repository.ts [hunk 8]
```diff
@@ -76,4 +78,14 @@ export class CertificateReadModelPostgresRepository<T>
             }
         });
     }
+
+    async getManyByBlockchainCertificateIds(
```
Comment (soanvig) [inline packages/origin-247-certificate/src/offchain-certificate/repositories/CertificateReadModel/CertificateReadModelPostgres.repository.ts:82]: unused

File: packages/origin-247-certificate/test/setup.ts [hunk 9]
```diff
@@ -123,11 +103,19 @@ export const bootstrapTestInstance: any = async () => {
...
+    );
 
     const connection = await app.resolve<Connection>(getConnectionToken());
     const tables = connection.entityMetadatas
         .filter(
-            (e) => e.tableName !== 'issuer_blockchain_properties' && e.tableName !== 'issuer_signer'
+            (e) =>
+                ![
+                    'issuer_blockchain_properties',
+                    'issuer_signer',
```
Comment (soanvig) [inline packages/origin-247-certificate/test/setup.ts:116]: these two can be dropped already, right?

**Review by soanvig — CHANGES_REQUESTED** [review]

File: packages/origin-247-certificate/src/offchain-certificate/utils/certificate.utils.ts (hunk 7 above)
Comment (JonaszJestem) [reply packages/origin-247-certificate/src/offchain-certificate/utils/certificate.utils.ts]: removed

File: packages/origin-247-certificate/src/offchain-certificate/repositories/CertificateReadModel/CertificateReadModelPostgres.repository.ts (hunk 8 above)
Comment (JonaszJestem) [reply packages/origin-247-certificate/src/offchain-certificate/repositories/CertificateReadModel/CertificateReadModelPostgres.repository.ts]: removed

File: packages/origin-247-certificate/src/offchain-certificate/validators/claim-command.validator.ts [hunk 10]
```diff
@@ -31,6 +29,42 @@ export const validateBatchClaimCommands = async (commands: IClaimCommand[]) => {
     throw validationErrors;
 };
 
+export function IsClaimData(validationOptions?: ValidationOptions) {
```
[the remark this answers is not in the document]
Comment (JonaszJestem) [reply packages/origin-247-certificate/src/offchain-certificate/validators/claim-command.validator.ts]: revmoed

File: packages/origin-247-certificate/test/setup.ts (hunk 9 above)
Comment (JonaszJestem) [reply packages/origin-247-certificate/test/setup.ts]: right, removed

**Review by patryk-kochanski — APPROVED** [review]

**Review by soanvig — APPROVED** [review]

**Pull request merged**